src/services/api.ts

```diff
@@ -19,9 +19,20 @@ export class Api {
 
     async joinRoom(
         roomId: RoomId,
-        password: string
+        participantName: string,
+        password: string | null
     ): Promise<JoinRoomResponse> {
-        return this.post(`/rooms/${roomId}/join`, { password })
+        try {
+            const joinResponse = await this.post(`/rooms/${roomId}/join`, {
+                participantName,
+                password,
+            })
+            return joinResponse
+        } catch (err) {
+            throw new Error(
+                'Connection error, maybe the room has not been created yet. Pls try again later'
```

Comment:
use API error message there.

Reply:
It was *Payment required*, should I use that, so that we'll change the text on the API side if needed ?

Reply:
I believe we can do that for the time being and check later on for the transation.

Reply:
I replaced as asked